### ISSUE TITLE
arrayvec 1.0

### ISSUE THREAD

**Issue by bluss** [bluss/arrayvec#48]
Planning for arrayvec 1.0

Prerequisites:

- Stable Manually Drop with good model for why keeping uninitialized data in arrayvec is safe.
- Settle API issues (Result vs Option and when to panic)

**Comment by johncf** [comment]
Please consider releasing a `0.4` after settling API issues raised in #45 and #46, before `1.0`. Would you like a PR?

**Comment by bluss** [comment]
I agree -- this is not actually for the next version, just an upcoming version.

**Comment by Limeth** [comment]
It might be worth considering `#[macro_export]`ing the `fix_array_impl_recursive` macro, too.

**Comment by bluss** [comment]
@Limeth I'd prefer to not do that, since it is useless -- it can't be used to impl `Array` for any array types, outside the crate.

**Comment by mqudsi** [comment]
With RFC2000 approved, I would hold off on a 1.0 until support for that lands in nightly. It can make this a zero-cost crate and provide compile-time overflow safety. The API will most definitely break, which is why I recommend not doing 1.0 until then (you'll be able to specify the length as a generic parameter, i.e. `ArrayVec<u8, 20>`).

https://github.com/rust-lang/rfcs/blob/master/text/2000-const-generics.md

**Comment by bluss** [comment]
@mqudsi The existing plan is that 1.0 is a good safety respecting version of the current functionality. That means ManuallyDrop/ManuallyUninit. Const generics is arrayvec 2.0.

**Comment by Luro02** [comment]
It would be useful, if you could create a [github milestone](https://github.com/bluss/arrayvec/milestones) and tag all the issues 🤗

**Comment by bluss** [comment]
Since we've been so slow to actually push out 1.0, const generics has now landed before 1.0 happened. So now it can be on the checklist. But we have every item ticked off, now.
